presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -195,4 +245,41 @@ private void initializeColumnReaders()
             columnReadersMap.put(column, ParquetColumnReader.createReader(column));
         }
     }
+
+    private Block readBlock(String name, Type type, List<String> path, IntList offsets)
+            throws IOException
+    {
+        path.add(name);
```

Comment:
any reason why you don't make this `path` variable (and the ones in other methods) immutable so that you don't need to do add/remove? I see similar add/remove logic around. And I am assuming these methods are accessed from a single thread, right?